### PR TITLE
Send formId & fileUrls to webhooks

### PR DESCRIPTION
**Pull request by mpinter**
- added formId
- moved the getFileIdsToUrl function to a util file so that it can be reused in multiple subservices
- mapping the file ids in form json by value their value (not by form schema)

### CODE REVIEW

**Review by erehulka — APPROVED** [review]

**Comment by mpinter** [comment]
Sorry for opening this early, I've intended to add the file stuff in as well - please recheck as that's a much larger change.

**Comment by MarekBodingerBA** [comment]
I don't really think it's good idea to replace the ids in the form data. Can't we just add the map and let them handle it please?

I would also add the slug of the form.

File: nest-forms-backend/src/utils/files.ts [hunk 1]
```diff
@@ -1,5 +1,50 @@
...
+
+import { FormWithFiles } from './types/prisma'
+
 // PDF_FORM_FAKE_FILE_ID is used to tell GinisService functions that they shouldn't look for this file in database
 export const PDF_FORM_FAKE_FILE_ID = 'pdf-form-fake-id'
 export const fileIdIsValid = (fileId: string): boolean =>
   fileId !== PDF_FORM_FAKE_FILE_ID
 export const PDF_EXPORT_FILE_NAME = 'pdf-export.pdf'
+
+export const getFileIdsToUrlMap = (
```
Comment (MarekBodingerBA) [inline nest-forms-backend/src/utils/files.ts:13]: Is there a reason no to extract it to FilesService? It would get all the variables (secret, self url) injected (which is the default way of getting them).

Comment (mpinter) [reply nest-forms-backend/src/utils/files.ts:13]: I guess it would be fine in FilesHelper, looking over it there are similar functions already. I'll move them so that I don't create more disparate approaches in codebase.

Not a fan though. On principle, I'd prefer having functions like this outside of services, injecting a couple config variables (as opposed to, for example, working with database or network calls) shouldn't suddenly stop a simple transformation from being a pure function.

Comment (mpinter) [reply nest-forms-backend/src/utils/files.ts:13]: Dug slightly deeper and talked in person afterwards. Since I don't want to delve into Files module refactor keeping this as is for now.

File: forms-shared/src/summary-email/renderSummaryEmail.tsx [hunk 2]
```diff
@@ -8,7 +8,7 @@ import { render } from '@react-email/components'
 import { SummaryEmail } from './SummaryEmail'
 import { FormDefinition } from '../definitions/formDefinitionTypes'
 
-export type RenderSummaryEmailFileIdUrlMap = Record<string, string>
+export type RenderSummaryEmailFileIdUrlMap = Record<string, { url: string; fileName: string }>
```
Comment (erehulka) [inline forms-shared/src/summary-email/renderSummaryEmail.tsx:11]: I would maybe change the naming to something like `RenderSummaryEmailFileIdInfoMap`, since now it contains file name as well.

File: nest-forms-backend/src/utils/__test__/files.spec.ts [hunk 3]
```diff
@@ -0,0 +1,74 @@
...
+    const mockJwtSecret = 'test-secret'
+
+    beforeEach(() => {
+      jest.clearAllMocks()
+      ;(jwt.sign as jest.Mock).mockReturnValue(mockJwtToken)
+    })
+
+    it('should create a map of file IDs to download URLs', () => {
+      const mockForm = {
+        files: [{ id: 'file1' }, { id: 'file2' }, { id: 'file3' }],
```
Comment (erehulka) [inline nest-forms-backend/src/utils/__test__/files.spec.ts:37]: maybe put some file names there for the sake of testing :D

**Review by erehulka — APPROVED** [review]
just two small comments, otherwise lgtm

File: nest-forms-backend/src/utils/files.ts [hunk 4]
```diff
@@ -1,5 +1,27 @@
...
+  form: FormWithFiles,
+  jwtSecret: string,
+  selfUrl: string,
+): Record<string, { url: string; fileName: string }> => {
```
Comment (erehulka) [inline nest-forms-backend/src/utils/files.ts:15]: Also one more thing: we can use the type `RenderSummaryEmailFileIdInfoMap` here, however since it is not tied only to the emails now, we can rename it to maybe just `FileIdInfoMap`. WDYT?

File: forms-shared/src/summary-email/renderSummaryEmail.tsx [hunk 5]
```diff
@@ -8,12 +8,12 @@ import { render } from '@react-email/components'
...
 import { FormDefinition } from '../definitions/formDefinitionTypes'
 
-export type RenderSummaryEmailFileIdUrlMap = Record<string, string>
+export type RenderSummaryEmailFileIdInfoMap = Record<string, { url: string; fileName: string }>
 
 export type RenderSummaryEmailPayload = {
   formDefinition: FormDefinition
   formData: GenericObjectType
-  fileIdUrlMap: RenderSummaryEmailFileIdUrlMap
+  fileIdUrlMap: RenderSummaryEmailFileIdInfoMap
```
Comment (erehulka) [inline forms-shared/src/summary-email/renderSummaryEmail.tsx:16]: regarding the naming change: please rename the key `fileIdUrlMap` as well.

**Review by erehulka — COMMENTED** [review]
Two more things about the naming

File: forms-shared/src/summary-email/renderSummaryEmail.tsx (hunk 5 above)
Comment (mpinter) [reply forms-shared/src/summary-email/renderSummaryEmail.tsx]: thanks - browsed around code after this, found a few more places where the 'fileIdUrlMap' was still used - it should be replaced everywhere where it became a map of id -> url,fileName. Kept it in one place in tests where it's still id->url map.

File: forms-shared/src/summary-email/SummaryEmail.tsx [hunk 6]
```diff
@@ -80,8 +80,8 @@ const StringValueRenderer = ({ value, isLast }: SummaryStringValueRendererProps)
 )
 
 const FileValueRenderer = ({ fileInfo, isLast }: SummaryFileValueRendererProps) => {
-  const fileIdUrlMap = useFileIdUrlMap()
-  const fileUrl = fileIdUrlMap[fileInfo.id].url
+  const fileIdInfoMap = useFileIdInfoMap()
+  const fileUrl = fileIdInfoMap[fileInfo.id].url
```
Comment (MarekBodingerBA) [inline forms-shared/src/summary-email/SummaryEmail.tsx:84]: Must be `fileIdInfoMap[fileInfo.id]?.url` or it will fail catastrophically in edge cases.

File: forms-shared/tests/summary-email/renderSummaryEmail.ts [hunk 7]
```diff
@@ -2,6 +2,7 @@ import { generatePageScreenshot } from '../../test-utils/generatePageScreenshot'
 import { toMatchImageSnapshot } from 'jest-image-snapshot'
 import { getExampleFormPairs } from '../../src/example-forms/getExampleFormPairs'
 import { renderSummaryEmail } from '../../src/summary-email/renderSummaryEmail'
+import { mapValues } from 'lodash'
```
Comment (MarekBodingerBA) [inline forms-shared/tests/summary-email/renderSummaryEmail.ts:5]: I try to keep lodash imports like this
`import mapValues from 'lodash/mapValues'

but we need eslint for this I think.

File: forms-shared/tests/summary-email/renderSummaryEmail.ts [hunk 8]
```diff
@@ -16,12 +17,16 @@ describe('renderSummaryEmail', () => {
           `https://example.com/${file.id}`,
         ])
         const fileIdUrlMap = Object.fromEntries(fileIdUrlMapEntries)
```
Comment (MarekBodingerBA) [inline forms-shared/tests/summary-email/renderSummaryEmail.ts:19]: I think we can remove `fileIdUrlMapEntries` and `fileIdUrlMap` then?

File: forms-shared/src/summary-email/SummaryEmail.tsx (hunk 6 above)
Comment (MarekBodingerBA) [reply forms-shared/src/summary-email/SummaryEmail.tsx:84]: FYI: https://www.typescriptlang.org/tsconfig/#noUncheckedIndexedAccess
